cmd/client-fs.go

```diff
@@ -338,12 +339,55 @@ func (f *fsClient) put(reader io.Reader, size int64, metadata map[string][]strin
 			err := f.toClientError(e, objectPath)
 			return totalWritten, err.Trace(objectPartPath, objectPath)
 		}
+
+		if len(metadata["mc-attrs"]) != 0 {
+			attr := parseAttribute(metadata["mc-attrs"][0])
+
+			mode, err := strconv.ParseUint(attr["mode"], 10, 32)
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+			// Change the mode of file
+			os.Chmod(objectPath, os.FileMode(mode))
+
+			uid, err := strconv.Atoi(attr["uid"])
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+
+			gid, err := strconv.Atoi(attr["uid"])
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+			// Change the owner
+			os.Chown(objectPath, uid, gid)
+
+			atime, err := strconv.ParseInt(attr["atime"], 10, 64)
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+
+			ctime, err := strconv.ParseInt(attr["ctime"], 10, 64)
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+			// Change the access, modify and change time
+			os.Chtimes(objectPath, time.Unix(atime, 0), time.Unix(ctime, 0))
```

Comment:
errors cannot be ignored for each os calls @sinhaashish you should handle them and return them upwards to the caller.